### PR TITLE
252@minor: Added support for AnimationEvent constructor arguments.

### PR DESCRIPTION
**Pull request by Atyn**

### CODE REVIEW

File: packages/happy-dom/src/event/events/AnimationEvent.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	public elapsedTime?: number;
+	public pseudoElement?: string;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param type Event type.
+	 * @param [eventInit] Event init.
+	 */
+	constructor(type: string, eventInit: IAnimationEventInit) {
```
Comment (capricorn86) [inline packages/happy-dom/src/event/events/AnimationEvent.ts:15]:
```suggestion
	constructor(type: string, eventInit?: IAnimationEventInit) {
```

File: packages/happy-dom/src/event/events/AnimationEvent.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Event from '../Event';
+import IAnimationEventInit from './IAnimationEventInit';
+
+export default class AnimationEvent extends Event {
+	public animationName?: string;
+	public elapsedTime?: number;
+	public pseudoElement?: string;
```
Comment (capricorn86) [inline packages/happy-dom/src/event/events/AnimationEvent.ts:7]:
```suggestion
	public animationName = '';
	public elapsedTime = 0;
	public pseudoElement = '';
```

File: packages/happy-dom/src/event/events/AnimationEvent.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	 *
+	 * @param type Event type.
+	 * @param [eventInit] Event init.
+	 */
+	constructor(type: string, eventInit: IAnimationEventInit) {
+		super(type, eventInit);
+		if (eventInit) {
+			this.animationName = eventInit.animationName;
+			this.elapsedTime = eventInit.elapsedTime;
+			this.pseudoElement = eventInit.pseudoElement;
```
Comment (capricorn86) [inline packages/happy-dom/src/event/events/AnimationEvent.ts:20]:
```suggestion
			this.animationName = eventInit?.animationName || '';
			this.elapsedTime = eventInit?.elapsedTime || 0;
			this.pseudoElement = eventInit?.pseudoElement || '';
```

File: packages/happy-dom/src/event/events/AnimationEvent.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import Event from '../Event';
+import IAnimationEventInit from './IAnimationEventInit';
+
+export default class AnimationEvent extends Event {
```
Comment (capricorn86) [inline packages/happy-dom/src/event/events/AnimationEvent.ts:4]: Can we add AnimationEvent to the main index.ts file as well?

**Review by capricorn86 — CHANGES_REQUESTED** [review]
Looks good! :+1: Just some minor things :slightly_smiling_face:

**Review by cyberBack — APPROVED** [review]

File: packages/happy-dom/src/event/events/AnimationEvent.ts (hunk 1 above)
Comment (Atyn) [reply packages/happy-dom/src/event/events/AnimationEvent.ts]: Fixed!

File: packages/happy-dom/src/event/events/AnimationEvent.ts (hunk 2 above)
Comment (Atyn) [reply packages/happy-dom/src/event/events/AnimationEvent.ts]: Fixed!

File: packages/happy-dom/src/event/events/AnimationEvent.ts (hunk 3 above)
Comment (Atyn) [reply packages/happy-dom/src/event/events/AnimationEvent.ts]: Fixed!

File: packages/happy-dom/src/event/events/AnimationEvent.ts (hunk 4 above)
Comment (Atyn) [reply packages/happy-dom/src/event/events/AnimationEvent.ts:4]: Fixed! 👍🙂

**Review by capricorn86 — APPROVED** [review]

**Comment by capricorn86** [comment]
Nice work @Atyn! :slightly_smiling_face:

**Pull request merged**